arm-trafficmanager/2017-05-01/swagger/trafficmanager.json

```diff
@@ -0,0 +1,972 @@
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Class representing a Traffic Manager profile."
+    },
+    "ProfileListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Profile"
+          },
+          "description": "Gets the list of Traffic manager profiles."
+        }
+      },
+      "description": "The list Traffic Manager profiles operation response."
+    },
+    "TrafficManagerNameAvailability": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The relative name."
+        },
+        "type": {
+          "type": "string",
+          "description": "Traffic Manager profile resource type."
+        },
+        "nameAvailable": {
+          "type": "boolean",
+          "description": "Describes whether the relative name is available or not."
+        },
+        "reason": {
+          "type": "string",
+          "description": "The reason why the name is not available, when applicable."
+        },
+        "message": {
+          "type": "string",
+          "description": "Descriptive message that explains why the name is not available, when applicable."
+        }
+      },
+      "description": "Class representing a Traffic Manager Name Availability response."
+    },
+    "Region": {
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "The code of the region"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the region"
+        },
+        "regions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Region"
+          },
+          "description": "The list of Regions grouped under this Region in the Geographic Hierarchy."
+        }
+      },
+      "description": "Class representing a region in the Geographic hierarchy used with the Geographic traffic routing method."
+    },
+    "GeographicHierarchyProperties": {
+      "properties": {
+        "geographicHierarchy": {
+          "$ref": "#/definitions/Region",
+          "description": "The region at the root of the hierarchy from all the regions in the hierarchy can be retrieved."
+        }
+      },
+      "description": "Class representing the properties of the Geographic hierarchy used with the Geographic traffic routing method."
+    },
+    "TrafficManagerGeographicHierarchy": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GeographicHierarchyProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Class representing the Geographic hierarchy used with the Geographic traffic routing method."
+    },
+    "Resource": {
```

Comment:
Please model `TrackedResource` as https://github.com/Azure/azure-rest-api-specs-pr/blob/master/common-types/resource-management/v1/types.json#L23 to avoid duplication and centralize model properties